From 4b4f53f8624c8ad780777fe7ceedf44087c596eb Mon Sep 17 00:00:00 2001
From: Basanagouda Koppad <basanagoudax.n.koppad@intel.com>
Date: Sun, 23 Jan 2022 19:21:29 +0530
Subject: [PATCH] CIV: Add android-removed-permissions.xml

Add unsupported features as unavailable
added in file android-removed-permissions.xml

Tracked-On: OAM-92946
Signed-off-by: Basanagouda Koppad <basanagoudax.n.koppad@intel.com>
Signed-off-by: Swee Yee Fonn <swee.yee.fonn@intel.com>

diff --git a/groups/device-specific/caas/android-removed-permissions.xml b/groups/device-specific/caas/android-removed-permissions.xml
new file mode 100644
index 0000000..18d3ef8
--- /dev/null
+++ b/groups/device-specific/caas/android-removed-permissions.xml
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (C) 2020 The Android Open Source Project
+
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+     You may obtain a copy of the License at
+
+          http://www.apache.org/licenses/LICENSE-2.0
+
+     Unless required by applicable law or agreed to in writing, software
+     distributed under the License is distributed on an "AS IS" BASIS,
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+-->
+
+<permissions>
+    <!-- Disable support for hardware we don't support -->
+    <unavailable-feature name="android.hardware.sensor.compass" />
+    <unavailable-feature name="android.hardware.sensor.accelerometer" />
+    <unavailable-feature name="android.hardware.sensor.gyroscope" />
+    <unavailable-feature name="android.hardware.sensor.light" />
+    <unavailable-feature name="android.hardware.sensor.ambient_temperature" />
+</permissions>
diff --git a/groups/device-specific/caas/caas.mk b/groups/device-specific/caas/caas.mk
index a3d92fb..d5f1c1b 100644
--- a/groups/device-specific/caas/caas.mk
+++ b/groups/device-specific/caas/caas.mk
@@ -24,7 +24,8 @@ PRODUCT_FULL_TREBLE_OVERRIDE := false
 {{/treble}}
 
 PRODUCT_DEFAULT_PROPERTY_OVERRIDES += ro.zygote=zygote64_32
-PRODUCT_COPY_FILES += system/core/rootdir/init.zygote64_32.rc:root/init.zygote64_32.rc
+PRODUCT_COPY_FILES += system/core/rootdir/init.zygote64_32.rc:root/init.zygote64_32.rc \
+    $(LOCAL_PATH)/android-removed-permissions.xml:system/etc/permissions/android-removed-permissions.xml
 
 BOARD_USE_64BIT_USERSPACE := true
 
diff --git a/groups/device-specific/caas/files.spec b/groups/device-specific/caas/files.spec
index c411bc1..463a243 100644
--- a/groups/device-specific/caas/files.spec
+++ b/groups/device-specific/caas/files.spec
@@ -11,6 +11,7 @@ overlay: "configurations for SystemUI"
 system.prop: "system properties file"
 {{ref_target}}.mk: "product definition file"
 r2_{{target}}.mk: "Ring 2 target for P.car"
+android-removed-permissions.xml: "remove features not supported"
 start_flash_usb.sh: "script for flashing qcow2 in Qemu"
 start_android_qcow2.sh: "script for starting android in Qemu"
 setup_host.sh: "script for setting up virtulization environment on host"
-- 
2.34.1

